test/check_sharding_data_test.lua

```diff
@@ -88,6 +88,47 @@ function g.test_sharding_key_ok()
 end
 
 
+function g.test_sharding_func_ok()
+    local space = table.deepcopy(test_space)
+    space.indexes = {primary_index}
+    local schema = {
+        spaces = {
+            space = space
+        }
+    }
+
+    -- strcrc32
+    g.space.sharding_func = 'strcrc32'
+    local res, err = ddl.check_schema(schema)
+    t.assert_equals(err, nil)
+    t.assert_equals(res, true)
+
+    -- vshard.router.bucket_id_strcrc32()
+    g.space.sharding_func = 'vshard.router.bucket_id_strcrc32()'
+    local res, err = ddl.check_schema(schema)
+    t.assert_equals(err, nil)
+    t.assert_equals(res, true)
```

Comment:
One can pass any custom function using 'dot.notation' so I would create a test for a custom sharding function and specify it using 'dot.notation'.

Reply:
Updated: in test `test_sharding_func_ok` in file `check_sharding_data_test.lua`:

```lua
-- sharding_func defined by user and set in _G
    local user_sharding_func = 'user_sharding_func'
    rawset(_G, user_sharding_func, {})
    g.space.sharding_func = user_sharding_func
    local res, err = ddl.check_schema(g.schema)
    t.assert_equals(err, nil)
    t.assert_equals(res, true)
```

In test `test_invalid_sharding_func` in file `check_sharding_data_test.lua` there is check that tests case with identifier (`mpcrc32`/`strcrc32`) and case with user function in `_G`:

```lua
-- unknown name
    g.space.sharding_func = 'bad_sharding_func'
    local ok, err = ddl.check_schema(g.schema)
    t.assert_equals(err,
        [[spaces["space"].sharding_func: unknown sharding_func "bad_sharding_func"]]
    )
    t.assert_equals(ok, nil)
```

In file `set_sharding_data_test.lua` test `test_ddl_sharding_func_space_user_func_name` was added

Reply:
Looks resolved. @ligurio?